lib/imagemounter/index.js

```diff
@@ -0,0 +1,120 @@
+import { BaseServicePlist } from '../base-service';
+import { fs } from '@appium/support';
+import B from 'bluebird';
+import log from '../logger';
+const { lstat, readFile, createReadStream } = fs;
+
+const MOBILE_IMAGE_MOUNTER_SERVICE_NAME = 'com.apple.mobile.mobile_image_mounter';
+const FILE_TYPE_IMAGE = 'image';
+const FILE_TYPE_SIGNATURE = 'signature';
+
+function checkIfError(ret) {
+    if (ret.Error) {
+        throw new Error(ret.Error);
+    }
+    return ret;
+}
+
+async function assertIsFile(filePath, fileType) {
+    /** @type {import('fs').Stats | undefined} */
+    let fileStat;
+    try {
+        fileStat = await lstat(filePath);
+    } catch (err) {
+        if (/** @type {NodeJS.ErrnoException} */(err).code === 'ENOENT') {
+            throw new Error(`The provided ${fileType} path does not exist: ${filePath}`);
+        }
+        throw err;
+    }
+    if (fileStat.isDirectory()) {
+        throw new Error(`The provided ${fileType} path is expected to be a file, but a directory was given: ${filePath}`);
+    }
+    return fileStat;
+}
+
+class ImageMounter extends BaseServicePlist {
+
+    /**
+     * Lookup for mounted images.
+     * @param {string} imageType Type of image, `Developer` by default.
+     * @returns {Promise<Buffer[]>} Signature of each mounted image.
+     */
+    async lookup(imageType = 'Developer') {
+        const ret = await this._plistService.sendPlistAndReceive({
+            Command: 'LookupImage',
+            ImageType: imageType
+        });
+        return checkIfError(ret).ImageSignature || [];
+    }
+
+    /**
+     * Check if developer image is mounted.
+     */
+    async isDeveloperImageMounted() {
+        return (await this.lookup()).length > 0;
+    }
+
+    /**
+     * Mount image for device.
+     * @param {string} imageFilePath The file path of image.
+     * @param {string} imageSignatureFilePath The signature file path of given `DeveloperDiskImage.dmg`
+     * @param {string} imageType Type of image, `Developer` by default.
+     */
+    async mount(imageFilePath, imageSignatureFilePath, imageType = 'Developer') {
+        //check file stats
+        const imageFileStat = await assertIsFile(imageFilePath, FILE_TYPE_IMAGE);
```

Comment:
this could be optimized:

```
const [imageFileStat] = await B.all([
  [imageFilePath, FILE_TYPE_IMAGE],
  [imageSignatureFilePath, FILE_TYPE_SIGNATURE]
].map((args) => assertIsFile(...args)))
```